src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,294 @@
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_LOW;
+	}
+
+	@Override
+	public int getCweId() {
+		return 311; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 4; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+		
+		if (getBaseMsg().getRequestHeader().getSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled()){
+				log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			}
+			System.out.println("\n\nBase isnt HTTP\n\n");
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		System.out.println(newRequest.getRequestHeader().toString());
+		try{
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			System.out.println(newRequest); //Checking the request
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			String record = "";
+			do {
+				sendAndReceive(newRequest, false);
+				System.out.println("\n\nRequest Succesful");
+				String loc = "Location: ";
+				String response = newRequest.getResponseHeader().toString();
+				System.out.println(response);
+				int ind = response.indexOf(loc);
+				String found = response.substring(ind+10, response.length());
+				System.out.println(ind+" Index");
+				if(ind > 0) {
+					System.out.println(found +" The string");
+					String[] lines = found.split("\r\n|\r|\n");
+				    String newHost = lines[0].substring(0, lines[0].length()-1);
+				    System.out.println(newHost);
+					String path = newRequest.getRequestHeader().getURI().getPath();
+					if(newHost.contains("https")){
+						newHost = newHost.substring(8, newHost.length());
+					    newRequest.getRequestHeader().setURI(new URI("https", null, newHost, 443, path));
+					    record = "https";
+					} else if(newHost.contains("http")) {
+						newHost = newHost.substring(7, newHost.length());
+					    newRequest.getRequestHeader().setURI(new URI("http", null, newHost, 80, path));
+					    record = "http";
+					}
+					System.out.println(newRequest.getRequestHeader().toString());
+					System.out.println("\n\nThe record is "+record);	
+				}
+				else {
+					if(record.equals("http")) {
+						System.out.println("It was http");
+						String newUri = newRequest.getRequestHeader().getURI().toString();
+						bingo(getRisk(), //Risk
```

Comment:
It should be extracted a method that raises the alert, to avoid code duplication.